<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to scCulturePredict • scCulturePredict</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to scCulturePredict">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.8.1/font/bootstrap-icons.css">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">scCulturePredict</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Functions</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/scCulturePredict-introduction.html">Introduction to scCulturePredict</a></li>
    <li><a class="dropdown-item" href="../articles/scCulturePredict-visualization.html">Visualization with scCulturePredict</a></li>
    <li><a class="dropdown-item" href="../articles/scCulturePredict-advanced.html">Advanced Usage of scCulturePredict</a></li>
  </ul>
</li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-news" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">News</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-news">
<li><h6 class="dropdown-header" data-toc-skip>Releases</h6></li>
    <li><a class="dropdown-item" href="../news/index.html#scculturepredict-0-99-0-2024-04-28">Version 0.99.0</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../news/index.html">Changelog</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/NCMBianchi/scCulturePredict" aria-label="GitHub repository"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to scCulturePredict</h1>
                        <h4 data-toc-skip class="author">Niccolò
Bianchi</h4>
            
            <h4 data-toc-skip class="date">2025-08-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NCMBianchi/scCulturePredict/blob/main/vignettes/scCulturePredict-introduction.Rmd" class="external-link"><code>vignettes/scCulturePredict-introduction.Rmd</code></a></small>
      <div class="d-none name"><code>scCulturePredict-introduction.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>scCulturePredict</code> is a comprehensive R package for
analyzing single-cell RNA-seq data with a focus on predicting cell
culture media from transcriptomic profiles. The package provides tools
for loading, preprocessing, and analyzing single-cell data using pathway
analysis and dimensionality reduction techniques.</p>
<p>This vignette provides an introduction to the core functionality of
the <code>scCulturePredict</code> package and demonstrates a basic
workflow using example data.</p>
<div class="section level3">
<h3 id="features">Features<a class="anchor" aria-label="anchor" href="#features"></a>
</h3>
<p>The <code>scCulturePredict</code> package offers several key
features:</p>
<ul>
<li>Data loading and preprocessing with robust error handling</li>
<li>Dimensionality reduction using UMAP</li>
<li>KEGG pathway analysis for biological interpretation</li>
<li>Prediction of cell culture media using similarity-based and machine
learning approaches</li>
<li>Evaluation and visualization of prediction results</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="section level3">
<h3 id="from-bioconductor">From Bioconductor<a class="anchor" aria-label="anchor" href="#from-bioconductor"></a>
</h3>
<p>The <code>scCulturePredict</code> package is available on
Bioconductor and can be installed using the following commands:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Package installation (for reference only - package should be installed separately):</span></span>
<span><span class="co"># if (!requireNamespace("BiocManager", quietly = TRUE))</span></span>
<span><span class="co">#     install.packages("BiocManager")</span></span>
<span><span class="co"># BiocManager::install("scCulturePredict")</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="from-github">From GitHub<a class="anchor" aria-label="anchor" href="#from-github"></a>
</h3>
<p>The development version can be installed directly from GitHub:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"nccb/scCulturePredict"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="getting-started">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<div class="section level3">
<h3 id="loading-the-package">Loading the Package<a class="anchor" aria-label="anchor" href="#loading-the-package"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://NCMBianchi.github.io/scCulturePredict">scCulturePredict</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example-data">Example Data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h3>
<p><code>scCulturePredict</code> comes with example data in two formats
that can be accessed using the <code><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file()</a></code> function:</p>
<ol style="list-style-type: decimal">
<li>CSV format - Simple comma-separated files with counts matrix and
metadata</li>
<li>10X Genomics format - Industry standard format with barcodes.tsv,
features.tsv, and matrix.mtx files</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the path to CSV format example data</span></span>
<span><span class="va">csv_data_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example_data"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get the path to 10X format example data</span></span>
<span><span class="va">tenx_data_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example_data_10x"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="basic-workflow">Basic Workflow<a class="anchor" aria-label="anchor" href="#basic-workflow"></a>
</h2>
<p>The typical workflow for using <code>scCulturePredict</code> consists
of the following steps:</p>
<ol style="list-style-type: decimal">
<li>Load single-cell data</li>
<li>Preprocess the data</li>
<li>Perform dimensionality reduction</li>
<li>Conduct pathway analysis</li>
<li>Predict cell culture media</li>
<li>Evaluate and visualize results</li>
</ol>
<div class="section level3">
<h3 id="quick-start-complete-analysis-with-one-function">Quick Start: Complete Analysis with One Function<a class="anchor" aria-label="anchor" href="#quick-start-complete-analysis-with-one-function"></a>
</h3>
<p>For most users, the easiest way to run the complete scCulturePredict
analysis is using the main <code>scumap()</code> function, which
performs all steps automatically:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Complete analysis with one function call</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">scumap</span><span class="op">(</span></span>
<span>    data_dir <span class="op">=</span> <span class="va">csv_data_dir</span>,</span>
<span>    kegg_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kegg"</span>, <span class="st">"example_pathway.keg"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span>,</span>
<span>    output_dir <span class="op">=</span> <span class="st">"./scumap_results"</span>,</span>
<span>    experiment_id <span class="op">=</span> <span class="st">"example_analysis"</span>,</span>
<span>    progress <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Show progress bar</span></span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span> <span class="co"># Print detailed messages</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Access the final results</span></span>
<span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">seurat_object</span></span>
<span><span class="va">pathway_results</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">pathway_results</span></span>
<span><span class="va">prediction_results</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">prediction_results</span></span>
<span><span class="va">evaluation_results</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">evaluation_results</span></span>
<span></span>
<span><span class="co"># View UMAP coordinates with predictions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">@</span><span class="va">meta.data</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP_1"</span>, <span class="st">"UMAP_2"</span>, <span class="st">"sample"</span>, <span class="st">"classification_pred"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check prediction accuracy</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">$</span><span class="va">sample</span>, <span class="va">seurat_object</span><span class="op">$</span><span class="va">classification_pred</span><span class="op">)</span></span></code></pre></div>
<p>The <code>scumap()</code> function automatically: - Loads and
validates your data - Performs preprocessing and quality control -
Generates UMAP coordinates - Conducts KEGG pathway analysis - Makes
predictions using both similarity and machine learning approaches -
Evaluates prediction performance - Creates publication-ready
visualizations - Saves all results to the specified output directory</p>
<p>All plots and results are automatically saved to
<code>./scumap_results/</code> for immediate use!</p>
</div>
<div class="section level3">
<h3 id="advanced-usage-step-by-step-workflow">Advanced Usage: Step-by-Step Workflow<a class="anchor" aria-label="anchor" href="#advanced-usage-step-by-step-workflow"></a>
</h3>
<p>For users who need more control over individual steps or want to
understand the underlying processes, the following sections demonstrate
each step in detail.</p>
</div>
<div class="section level3">
<h3 id="load-data">1. Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h3>
<p>The <code>scCulturePredict</code> package can load data in both CSV
and 10X Genomics formats. Here are examples of both approaches:</p>
<div class="section level4">
<h4 id="loading-csv-format-data">Loading CSV Format Data<a class="anchor" aria-label="anchor" href="#loading-csv-format-data"></a>
</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the CSV format example data</span></span>
<span><span class="va">seurat_object_csv</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_data.html">load_data</a></span><span class="op">(</span></span>
<span>    data_dir <span class="op">=</span> <span class="va">csv_data_dir</span>,</span>
<span>    experiment_id <span class="op">=</span> <span class="st">"example"</span>,</span>
<span>    min_cells <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    min_features <span class="op">=</span> <span class="fl">10</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="loading-10x-genomics-format-data">Loading 10X Genomics Format Data<a class="anchor" aria-label="anchor" href="#loading-10x-genomics-format-data"></a>
</h4>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the 10X format example data</span></span>
<span><span class="va">seurat_object_10x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_data.html">load_data</a></span><span class="op">(</span></span>
<span>    data_dir <span class="op">=</span> <span class="va">tenx_data_dir</span>,</span>
<span>    experiment_id <span class="op">=</span> <span class="st">"example_10x"</span>,</span>
<span>    min_cells <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    min_features <span class="op">=</span> <span class="fl">10</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Alternatively, you can use Seurat's Read10X function directly</span></span>
<span><span class="va">tenx_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X.html" class="external-link">Read10X</a></span><span class="op">(</span><span class="va">tenx_data_dir</span><span class="op">)</span></span>
<span><span class="va">seurat_object_10x_direct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span></span>
<span>    counts <span class="op">=</span> <span class="va">tenx_data</span>,</span>
<span>    min.cells <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    min.features <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="preprocess-data">2. Preprocess Data<a class="anchor" aria-label="anchor" href="#preprocess-data"></a>
</h3>
<p>Next, we preprocess the data, which includes normalization, scaling,
and identifying variable features:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Preprocess the data</span></span>
<span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocess_data.html">preprocess_data</a></span><span class="op">(</span></span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>    normalization_method <span class="op">=</span> <span class="st">"LogNormalize"</span>,</span>
<span>    scale_factor <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>    variable_features_n <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="perform-dimensionality-reduction">3. Perform Dimensionality Reduction<a class="anchor" aria-label="anchor" href="#perform-dimensionality-reduction"></a>
</h3>
<p>We then reduce the dimensionality of the data using UMAP:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform dimensionality reduction</span></span>
<span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/reduce_dimensions.html">reduce_dimensions</a></span><span class="op">(</span></span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>    pca_dims <span class="op">=</span> <span class="fl">30</span>,</span>
<span>    umap_dims <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Let’s visualize the UMAP embedding:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create UMAP visualization</span></span>
<span><span class="va">umap_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_umap_plot.html">create_umap_plot</a></span><span class="op">(</span></span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>    color_by <span class="op">=</span> <span class="st">"sample"</span>,</span>
<span>    point_size <span class="op">=</span> <span class="fl">1.2</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"UMAP of Example Data"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Display the plot</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">umap_plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="scCulturePredict-introduction_files/figure-html/umap_vis-1.png" width="768"></p>
</div>
<div class="section level3">
<h3 id="pathway-analysis">4. Pathway Analysis<a class="anchor" aria-label="anchor" href="#pathway-analysis"></a>
</h3>
<p>Now we perform KEGG pathway analysis to identify biological pathways
that are active in our data:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Path to KEGG data</span></span>
<span><span class="va">kegg_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kegg"</span>, <span class="st">"example_pathway.keg"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Parse KEGG data</span></span>
<span><span class="va">kegg_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/parse_kegg_keg.html">parse_kegg_keg</a></span><span class="op">(</span><span class="va">kegg_file</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Build pathway fingerprints</span></span>
<span><span class="va">pathway_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_fingerprints.html">build_fingerprints</a></span><span class="op">(</span></span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>    kegg_data <span class="op">=</span> <span class="va">kegg_data</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="predict-cell-culture-media">5. Predict Cell Culture Media<a class="anchor" aria-label="anchor" href="#predict-cell-culture-media"></a>
</h3>
<p>We can now predict the culture media using both similarity-based and
SVM approaches:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Similarity-based prediction</span></span>
<span><span class="va">similarity_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predict_by_similarity.html">predict_by_similarity</a></span><span class="op">(</span></span>
<span>    pathway_matrix <span class="op">=</span> <span class="va">pathway_matrix</span>,</span>
<span>    signature_matrix <span class="op">=</span> <span class="va">signature_matrix</span>,</span>
<span>    threshold <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># SVM-based prediction</span></span>
<span><span class="va">svm_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predict_by_svm.html">predict_by_svm</a></span><span class="op">(</span></span>
<span>    pathway_matrix <span class="op">=</span> <span class="va">pathway_matrix</span>,</span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="evaluate-and-visualize-results">6. Evaluate and Visualize Results<a class="anchor" aria-label="anchor" href="#evaluate-and-visualize-results"></a>
</h3>
<p>First, let’s add some mock prediction results to demonstrate the
evaluation functionality:</p>
<p>Finally, we evaluate the prediction results and create
visualizations:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Evaluate predictions</span></span>
<span><span class="va">evaluation_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate_predictions.html">evaluate_predictions</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="create-accuracy-plot">Create accuracy plot<a class="anchor" aria-label="anchor" href="#create-accuracy-plot"></a>
</h2>
<p>accuracy_plot &lt;- create_accuracy_plot( evaluation_results =
evaluation_results, method = “svm”, title = “SVM Prediction Accuracy”
)</p>
</div>
<div class="section level2">
<h2 id="display-the-plot">Display the plot<a class="anchor" aria-label="anchor" href="#display-the-plot"></a>
</h2>
<p>print(accuracy_plot)</p>
</div>
<div class="section level2">
<h2 id="create-confusion-matrix-heatmap">Create confusion matrix heatmap<a class="anchor" aria-label="anchor" href="#create-confusion-matrix-heatmap"></a>
</h2>
<p>confusion_plot &lt;- create_confusion_heatmap( evaluation_results =
evaluation_results, method = “direct”, title = “Direct Prediction
Confusion Matrix” )</p>
</div>
<div class="section level2">
<h2 id="display-the-plot-1">Display the plot<a class="anchor" aria-label="anchor" href="#display-the-plot-1"></a>
</h2>
<p>print(confusion_plot)</p>
<pre><code>
# Saving Results

You can save all visualization plots to a directory:


``` r
# Create directory for results
results_dir &lt;- "scCulturePredict_results"
if (!dir.exists(results_dir)) dir.create(results_dir)

# Save visualization plots
save_visualization_plots(
    seurat_object = seurat_object,
    evaluation_results = evaluation_results,
    output_dir = results_dir,
    prefix = "example",
    verbose = TRUE
)</code></pre>
</div>
<div class="section level2">
<h2 id="working-with-different-data-formats">Working with Different Data Formats<a class="anchor" aria-label="anchor" href="#working-with-different-data-formats"></a>
</h2>
<div class="section level3">
<h3 id="csv-format">CSV Format<a class="anchor" aria-label="anchor" href="#csv-format"></a>
</h3>
<p>The CSV format is simpler and consists of: - A counts matrix with
genes as rows and cells as columns - A metadata file with cells as rows
and annotations as columns</p>
<p>This format is easy to create and manipulate using standard R
functions like <code><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv()</a></code> and <code><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv()</a></code>.</p>
</div>
<div class="section level3">
<h3 id="x-genomics-format">10X Genomics Format<a class="anchor" aria-label="anchor" href="#x-genomics-format"></a>
</h3>
<p>The 10X Genomics format is an industry standard and consists of: -
<code>barcodes.tsv</code> - Contains cell barcodes -
<code>features.tsv</code> - Contains gene/feature information -
<code>matrix.mtx</code> - Contains the count matrix in sparse format</p>
<p>This format is more memory-efficient for large datasets and is widely
used in single-cell analysis.</p>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>This vignette has demonstrated the basic workflow for using the
<code>scCulturePredict</code> package to analyze single-cell RNA-seq
data and predict cell culture media, with support for both CSV and 10X
Genomics data formats. For more advanced usage and customization
options, please refer to the advanced vignette and the function
documentation.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ggplot2_3.5.2           Seurat_5.3.0            SeuratObject_5.1.0     </span></span>
<span><span class="co">#&gt; [4] sp_2.2-0                scCulturePredict_0.99.9 BiocStyle_2.36.0       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] deldir_2.0-4           pbapply_1.7-4          gridExtra_2.3         </span></span>
<span><span class="co">#&gt;   [4] rlang_1.1.6            magrittr_2.0.3         RcppAnnoy_0.0.22      </span></span>
<span><span class="co">#&gt;   [7] spatstat.geom_3.5-0    matrixStats_1.5.0      ggridges_0.5.6        </span></span>
<span><span class="co">#&gt;  [10] compiler_4.5.1         png_0.1-8              systemfonts_1.2.3     </span></span>
<span><span class="co">#&gt;  [13] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.1         </span></span>
<span><span class="co">#&gt;  [16] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        </span></span>
<span><span class="co">#&gt;  [19] promises_1.3.3         rmarkdown_2.29         ragg_1.4.0            </span></span>
<span><span class="co">#&gt;  [22] purrr_1.1.0            xfun_0.52              cachem_1.1.0          </span></span>
<span><span class="co">#&gt;  [25] jsonlite_2.0.0         goftest_1.2-3          later_1.4.2           </span></span>
<span><span class="co">#&gt;  [28] spatstat.utils_3.1-5   irlba_2.3.5.1          parallel_4.5.1        </span></span>
<span><span class="co">#&gt;  [31] cluster_2.1.8.1        R6_2.6.1               ica_1.0-3             </span></span>
<span><span class="co">#&gt;  [34] spatstat.data_3.1-6    stringi_1.8.7          bslib_0.9.0           </span></span>
<span><span class="co">#&gt;  [37] RColorBrewer_1.1-3     reticulate_1.43.0      spatstat.univar_3.1-4 </span></span>
<span><span class="co">#&gt;  [40] parallelly_1.45.1      lmtest_0.9-40          jquerylib_0.1.4       </span></span>
<span><span class="co">#&gt;  [43] scattermore_1.2        Rcpp_1.1.0             bookdown_0.43         </span></span>
<span><span class="co">#&gt;  [46] knitr_1.50             tensor_1.5.1           future.apply_1.20.0   </span></span>
<span><span class="co">#&gt;  [49] zoo_1.8-14             sctransform_0.4.2      httpuv_1.6.16         </span></span>
<span><span class="co">#&gt;  [52] Matrix_1.7-3           splines_4.5.1          igraph_2.1.4          </span></span>
<span><span class="co">#&gt;  [55] tidyselect_1.2.1       abind_1.4-8            yaml_2.3.10           </span></span>
<span><span class="co">#&gt;  [58] spatstat.random_3.4-1  spatstat.explore_3.5-2 codetools_0.2-20      </span></span>
<span><span class="co">#&gt;  [61] miniUI_0.1.2           listenv_0.9.1          plyr_1.8.9            </span></span>
<span><span class="co">#&gt;  [64] lattice_0.22-7         tibble_3.3.0           withr_3.0.2           </span></span>
<span><span class="co">#&gt;  [67] shiny_1.11.1           ROCR_1.0-11            evaluate_1.0.4        </span></span>
<span><span class="co">#&gt;  [70] Rtsne_0.17             future_1.67.0          fastDummies_1.7.5     </span></span>
<span><span class="co">#&gt;  [73] desc_1.4.3             survival_3.8-3         polyclip_1.10-7       </span></span>
<span><span class="co">#&gt;  [76] fitdistrplus_1.2-4     pillar_1.11.0          BiocManager_1.30.26   </span></span>
<span><span class="co">#&gt;  [79] KernSmooth_2.23-26     plotly_4.11.0          generics_0.1.4        </span></span>
<span><span class="co">#&gt;  [82] RcppHNSW_0.6.0         scales_1.4.0           globals_0.18.0        </span></span>
<span><span class="co">#&gt;  [85] xtable_1.8-4           glue_1.8.0             lazyeval_0.2.2        </span></span>
<span><span class="co">#&gt;  [88] tools_4.5.1            data.table_1.17.8      RSpectra_0.16-2       </span></span>
<span><span class="co">#&gt;  [91] RANN_2.6.2             fs_1.6.6               dotCall64_1.2         </span></span>
<span><span class="co">#&gt;  [94] cowplot_1.2.0          grid_4.5.1             tidyr_1.3.1           </span></span>
<span><span class="co">#&gt;  [97] nlme_3.1-168           patchwork_1.3.1        cli_3.6.5             </span></span>
<span><span class="co">#&gt; [100] spatstat.sparse_3.1-0  textshaping_1.0.1      spam_2.11-1           </span></span>
<span><span class="co">#&gt; [103] viridisLite_0.4.2      dplyr_1.1.4            uwot_0.2.3            </span></span>
<span><span class="co">#&gt; [106] gtable_0.3.6           sass_0.4.10            digest_0.6.37         </span></span>
<span><span class="co">#&gt; [109] progressr_0.15.1       ggrepel_0.9.6          htmlwidgets_1.6.4     </span></span>
<span><span class="co">#&gt; [112] farver_2.1.2           htmltools_0.5.8.1      pkgdown_2.1.3         </span></span>
<span><span class="co">#&gt; [115] lifecycle_1.0.4        httr_1.4.7             mime_0.13             </span></span>
<span><span class="co">#&gt; [118] MASS_7.3-65</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Niccolò Bianchi.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
