<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Advanced Usage of scCulturePredict • scCulturePredict</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Advanced Usage of scCulturePredict">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.8.1/font/bootstrap-icons.css">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">scCulturePredict</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.30</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Functions</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><a class="dropdown-item" href="../articles/scCulturePredict-introduction.html">Introduction to scCulturePredict</a></li>
    <li><a class="dropdown-item" href="../articles/scCulturePredict-visualization.html">Visualization with scCulturePredict</a></li>
    <li><a class="dropdown-item" href="../articles/scCulturePredict-advanced.html">Advanced Usage of scCulturePredict</a></li>
  </ul>
</li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-news" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">News</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-news">
<li><h6 class="dropdown-header" data-toc-skip>Releases</h6></li>
    <li><a class="dropdown-item" href="../news/index.html#scculturepredict-0-99-0-2024-04-28">Version 0.99.0</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><a class="dropdown-item" href="../news/index.html">Changelog</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/NCMBianchi/scCulturePredict" aria-label="GitHub repository"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Advanced Usage of scCulturePredict</h1>
                        <h4 data-toc-skip class="author">Niccolò
Bianchi</h4>
            
            <h4 data-toc-skip class="date">2025-08-27</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/NCMBianchi/scCulturePredict/blob/main/vignettes/scCulturePredict-advanced.Rmd" class="external-link"><code>vignettes/scCulturePredict-advanced.Rmd</code></a></small>
      <div class="d-none name"><code>scCulturePredict-advanced.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette covers advanced usage of the
<code>scCulturePredict</code> package, building on the basic concepts
introduced in the introductory vignette. Here, we explore more
sophisticated analyses, customization options, and performance
optimizations.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://NCMBianchi.github.io/scCulturePredict">scCulturePredict</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<p>library(ggplot2) library(parallel) library(doParallel)
library(foreach)</p>
<pre><code>
# Setup Example Data

We'll use the same example data as in the introduction vignette, but we'll process it differently:


``` r
# Get the path to example data
tenx_data_dir &lt;- system.file("extdata", "example_data", package = "scCulturePredict")

# For demonstration purposes, create a mock Seurat object
counts &lt;- read.csv(file.path(tenx_data_dir, "counts.csv"), row.names = 1)
metadata &lt;- read.csv(file.path(tenx_data_dir, "metadata.csv"), row.names = 1)
seurat_object &lt;- CreateSeuratObject(counts = as.matrix(counts), meta.data = metadata)</code></pre>
</div>
<div class="section level2">
<h2 id="advanced-scculture-usage">Advanced scCulture() Usage<a class="anchor" aria-label="anchor" href="#advanced-scculture-usage"></a>
</h2>
<p>The <code><a href="../reference/scCulture.html">scCulture()</a></code> function provides several advanced
options for power users who need more control over the analysis pipeline
or want to optimize performance for large datasets.</p>
<div class="section level3">
<h3 id="progress-tracking-and-parallel-processing">Progress Tracking and Parallel Processing<a class="anchor" aria-label="anchor" href="#progress-tracking-and-parallel-processing"></a>
</h3>
<p>For large datasets, you can enable progress tracking and parallel
processing to monitor and speed up the analysis:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Advanced usage with all options</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scCulture.html">scCulture</a></span><span class="op">(</span></span>
<span>    tenx_data_dir <span class="op">=</span> <span class="va">tenx_data_dir</span>,</span>
<span>    input_type <span class="op">=</span> <span class="st">"10x"</span>,</span>
<span>    kegg_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kegg"</span>, <span class="st">"example_pathway.keg"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span>,</span>
<span>    output_dir <span class="op">=</span> <span class="st">"./advanced_results"</span>,</span>
<span>    mode <span class="op">=</span> <span class="st">"build"</span>,</span>
<span>    experiment_id <span class="op">=</span> <span class="st">"advanced_analysis"</span>,</span>
<span>    progress <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Show detailed progress bar</span></span>
<span>    parallel <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Enable parallel processing</span></span>
<span>    n_cores <span class="op">=</span> <span class="fl">4</span>, <span class="co"># Use 4 CPU cores</span></span>
<span>    perform_tsne <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Include t-SNE analysis</span></span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span> <span class="co"># Detailed progress messages</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The function returns comprehensive results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">results</span>, max.level <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="customizing-analysis-parameters">Customizing Analysis Parameters<a class="anchor" aria-label="anchor" href="#customizing-analysis-parameters"></a>
</h3>
<p>You can customize various aspects of the analysis by modifying the
underlying functions:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example: Custom analysis with modified parameters</span></span>
<span><span class="va">results_custom</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scCulture.html">scCulture</a></span><span class="op">(</span></span>
<span>    tenx_data_dir <span class="op">=</span> <span class="va">tenx_data_dir</span>,</span>
<span>    input_type <span class="op">=</span> <span class="st">"10x"</span>,</span>
<span>    kegg_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kegg"</span>, <span class="st">"example_pathway.keg"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span>,</span>
<span>    output_dir <span class="op">=</span> <span class="st">"./custom_results"</span>,</span>
<span>    mode <span class="op">=</span> <span class="st">"build"</span>,</span>
<span>    experiment_id <span class="op">=</span> <span class="st">"custom_analysis"</span>,</span>
<span>    progress <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Access and modify the Seurat object</span></span>
<span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="va">results_custom</span><span class="op">$</span><span class="va">seurat_object</span></span>
<span></span>
<span><span class="co"># You can then run additional custom analyses on the results</span></span>
<span><span class="co"># For example, additional clustering or custom visualizations</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="batch-processing-multiple-datasets">Batch Processing Multiple Datasets<a class="anchor" aria-label="anchor" href="#batch-processing-multiple-datasets"></a>
</h3>
<p>For processing multiple datasets, you can create a wrapper
function:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Function to process multiple datasets</span></span>
<span><span class="va">process_multiple_datasets</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">datasets</span>, <span class="va">kegg_file</span>, <span class="va">base_output_dir</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">results_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span>    <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">datasets</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="va">dataset_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">datasets</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span></span>
<span>        <span class="va">data_path</span> <span class="op">&lt;-</span> <span class="va">datasets</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"Processing dataset %d/%d: %s\n"</span>, <span class="va">i</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">datasets</span><span class="op">)</span>, <span class="va">dataset_name</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>        <span class="co"># Create unique output directory for each dataset</span></span>
<span>        <span class="va">output_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">base_output_dir</span>, <span class="va">dataset_name</span><span class="op">)</span></span>
<span></span>
<span>        <span class="co"># Run scCulture analysis</span></span>
<span>        <span class="va">results_list</span><span class="op">[[</span><span class="va">dataset_name</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scCulture.html">scCulture</a></span><span class="op">(</span></span>
<span>            tenx_data_dir <span class="op">=</span> <span class="va">data_path</span>,</span>
<span>            input_type <span class="op">=</span> <span class="st">"10x"</span>,</span>
<span>            kegg_file <span class="op">=</span> <span class="va">kegg_file</span>,</span>
<span>            output_dir <span class="op">=</span> <span class="va">output_dir</span>,</span>
<span>            mode <span class="op">=</span> <span class="st">"build"</span>,</span>
<span>            experiment_id <span class="op">=</span> <span class="va">dataset_name</span>,</span>
<span>            progress <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Disable progress bar for batch processing</span></span>
<span>            parallel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            verbose <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># Reduce verbosity for batch processing</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">results_list</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Example usage (not run)</span></span>
<span><span class="co"># datasets &lt;- list(</span></span>
<span><span class="co">#   "experiment_1" = "/path/to/data1",</span></span>
<span><span class="co">#   "experiment_2" = "/path/to/data2",</span></span>
<span><span class="co">#   "experiment_3" = "/path/to/data3"</span></span>
<span><span class="co"># )</span></span>
<span><span class="co">#</span></span>
<span><span class="co"># batch_results &lt;- process_multiple_datasets(</span></span>
<span><span class="co">#   datasets = datasets,</span></span>
<span><span class="co">#   kegg_file = "pathway_file.keg",</span></span>
<span><span class="co">#   base_output_dir = "./batch_results"</span></span>
<span><span class="co"># )</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="advanced-preprocessing">Advanced Preprocessing<a class="anchor" aria-label="anchor" href="#advanced-preprocessing"></a>
</h2>
<div class="section level3">
<h3 id="batch-effect-correction">Batch Effect Correction<a class="anchor" aria-label="anchor" href="#batch-effect-correction"></a>
</h3>
<p>Batch effects can significantly impact single-cell analyses. Here, we
demonstrate how to correct for batch effects:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Identify batch variable in metadata</span></span>
<span><span class="va">batch_var</span> <span class="op">&lt;-</span> <span class="st">"batch"</span></span>
<span></span>
<span><span class="co"># Run integration to correct for batch effects</span></span>
<span><span class="va">seurat_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SplitObject.html" class="external-link">SplitObject</a></span><span class="op">(</span><span class="va">seurat_object</span>, split.by <span class="op">=</span> <span class="va">batch_var</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Normalize and identify variable features for each dataset independently</span></span>
<span><span class="va">seurat_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">seurat_list</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>    <span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">x</span>, selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Select integration features</span></span>
<span><span class="va">features</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SelectIntegrationFeatures.html" class="external-link">SelectIntegrationFeatures</a></span><span class="op">(</span>object.list <span class="op">=</span> <span class="va">seurat_list</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Find integration anchors</span></span>
<span><span class="va">anchors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindIntegrationAnchors.html" class="external-link">FindIntegrationAnchors</a></span><span class="op">(</span></span>
<span>    object.list <span class="op">=</span> <span class="va">seurat_list</span>,</span>
<span>    anchor.features <span class="op">=</span> <span class="va">features</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Integrate data</span></span>
<span><span class="va">seurat_integrated</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/IntegrateData.html" class="external-link">IntegrateData</a></span><span class="op">(</span>anchorset <span class="op">=</span> <span class="va">anchors</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Switch to integrated assay for downstream analysis</span></span>
<span><span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">seurat_integrated</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"integrated"</span></span>
<span></span>
<span><span class="co"># Standard preprocessing on integrated data</span></span>
<span><span class="va">seurat_integrated</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">seurat_integrated</span><span class="op">)</span></span>
<span><span class="va">seurat_integrated</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">seurat_integrated</span><span class="op">)</span></span>
<span><span class="va">seurat_integrated</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">seurat_integrated</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize results with batch variable</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_integrated</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="va">batch_var</span><span class="op">)</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_integrated</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="st">"sample"</span><span class="op">)</span></span>
<span><span class="va">p1</span> <span class="op">+</span> <span class="va">p2</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="custom-qc-and-filtering">Custom QC and Filtering<a class="anchor" aria-label="anchor" href="#custom-qc-and-filtering"></a>
</h3>
<p>More stringent QC can improve downstream analyses:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate QC metrics</span></span>
<span><span class="va">seurat_object</span><span class="op">[[</span><span class="st">"percent_mito"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/PercentageFeatureSet.html" class="external-link">PercentageFeatureSet</a></span><span class="op">(</span><span class="va">seurat_object</span>, pattern <span class="op">=</span> <span class="st">"^MT-"</span><span class="op">)</span></span>
<span><span class="va">seurat_object</span><span class="op">[[</span><span class="st">"percent_ribo"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/PercentageFeatureSet.html" class="external-link">PercentageFeatureSet</a></span><span class="op">(</span><span class="va">seurat_object</span>, pattern <span class="op">=</span> <span class="st">"^RP[SL]"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize QC metrics</span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/VlnPlot.html" class="external-link">VlnPlot</a></span><span class="op">(</span><span class="va">seurat_object</span>,</span>
<span>    features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"nFeature_RNA"</span>, <span class="st">"nCount_RNA"</span>, <span class="st">"percent_mito"</span>, <span class="st">"percent_ribo"</span><span class="op">)</span>,</span>
<span>    ncol <span class="op">=</span> <span class="fl">4</span>, pt.size <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Custom filtering</span></span>
<span><span class="va">seurat_filtered</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">seurat_object</span>,</span>
<span>    subset <span class="op">=</span> <span class="va">nFeature_RNA</span> <span class="op">&gt;</span> <span class="fl">500</span> <span class="op">&amp;</span></span>
<span>        <span class="va">nFeature_RNA</span> <span class="op">&lt;</span> <span class="fl">5000</span> <span class="op">&amp;</span></span>
<span>        <span class="va">percent_mito</span> <span class="op">&lt;</span> <span class="fl">15</span> <span class="op">&amp;</span></span>
<span>        <span class="va">percent_ribo</span> <span class="op">&gt;</span> <span class="fl">5</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="advanced-kegg-pathway-analysis">Advanced KEGG Pathway Analysis<a class="anchor" aria-label="anchor" href="#advanced-kegg-pathway-analysis"></a>
</h2>
<div class="section level3">
<h3 id="custom-pathway-definition">Custom Pathway Definition<a class="anchor" aria-label="anchor" href="#custom-pathway-definition"></a>
</h3>
<p>You can define custom pathways beyond standard KEGG pathways:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define custom pathway gene sets</span></span>
<span><span class="va">custom_pathways</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    <span class="st">"Custom_Pathway_1"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSG00000001"</span>, <span class="st">"ENSG00000002"</span>, <span class="st">"ENSG00000003"</span><span class="op">)</span>,</span>
<span>    <span class="st">"Custom_Pathway_2"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSG00000004"</span>, <span class="st">"ENSG00000005"</span>, <span class="st">"ENSG00000006"</span><span class="op">)</span>,</span>
<span>    <span class="st">"Custom_Pathway_3"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSG00000007"</span>, <span class="st">"ENSG00000008"</span>, <span class="st">"ENSG00000009"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert to format compatible with build_fingerprints</span></span>
<span><span class="va">custom_pathways_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    pathway_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">custom_pathways</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">custom_pathways</span>, <span class="va">length</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">custom_pathways</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Build custom pathway fingerprints</span></span>
<span><span class="va">custom_fingerprints</span> <span class="op">&lt;-</span> <span class="fu">build_custom_fingerprints</span><span class="op">(</span></span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>    pathway_gene_map <span class="op">=</span> <span class="va">custom_pathways_df</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="weighted-pathway-analysis">Weighted Pathway Analysis<a class="anchor" aria-label="anchor" href="#weighted-pathway-analysis"></a>
</h3>
<p>Incorporate gene importance weights in pathway analysis:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define gene weights (e.g., from differential expression analysis)</span></span>
<span><span class="va">gene_weights</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>    gene_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span>,</span>
<span>    weight <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Build weighted pathway fingerprints</span></span>
<span><span class="va">weighted_fingerprints</span> <span class="op">&lt;-</span> <span class="fu">build_weighted_fingerprints</span><span class="op">(</span></span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>    kegg_data <span class="op">=</span> <span class="va">kegg_data</span>,</span>
<span>    gene_weights <span class="op">=</span> <span class="va">gene_weights</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="performance-optimization">Performance Optimization<a class="anchor" aria-label="anchor" href="#performance-optimization"></a>
</h2>
<div class="section level3">
<h3 id="parallel-processing">Parallel Processing<a class="anchor" aria-label="anchor" href="#parallel-processing"></a>
</h3>
<p>For large datasets, parallel processing can significantly improve
performance:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Set up parallel backend</span></span>
<span><span class="va">num_cores</span> <span class="op">&lt;-</span> <span class="fu">detectCores</span><span class="op">(</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span></span>
<span><span class="fu">registerDoParallel</span><span class="op">(</span>cores <span class="op">=</span> <span class="va">num_cores</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use parallel processing in pathway analysis</span></span>
<span><span class="va">pathway_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_fingerprints.html">build_fingerprints</a></span><span class="op">(</span></span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>    kegg_data <span class="op">=</span> <span class="va">kegg_data</span>,</span>
<span>    use_parallel <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    num_cores <span class="op">=</span> <span class="va">num_cores</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="memory-optimization">Memory Optimization<a class="anchor" aria-label="anchor" href="#memory-optimization"></a>
</h3>
<p>When working with large datasets, memory optimization is crucial:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Process data in chunks</span></span>
<span><span class="va">chunk_size</span> <span class="op">&lt;-</span> <span class="fl">1000</span> <span class="co"># cells per chunk</span></span>
<span><span class="va">cell_chunks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">ceiling</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span> <span class="va">chunk_size</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Process each chunk</span></span>
<span><span class="va">results_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">cell_chunks</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">verbose</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"Processing chunk %d of %d"</span>, <span class="va">i</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">cell_chunks</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># Subset data</span></span>
<span>    <span class="va">chunk_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">seurat_object</span>, cells <span class="op">=</span> <span class="va">cell_chunks</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># Process chunk</span></span>
<span>    <span class="va">chunk_results</span> <span class="op">&lt;-</span> <span class="fu">process_chunk</span><span class="op">(</span><span class="va">chunk_obj</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># Store results</span></span>
<span>    <span class="va">results_list</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">chunk_results</span></span>
<span></span>
<span>    <span class="co"># Explicit garbage collection</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/gc.html" class="external-link">gc</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Combine results</span></span>
<span><span class="va">combined_results</span> <span class="op">&lt;-</span> <span class="fu">combine_chunk_results</span><span class="op">(</span><span class="va">results_list</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="advanced-prediction-models">Advanced Prediction Models<a class="anchor" aria-label="anchor" href="#advanced-prediction-models"></a>
</h2>
<div class="section level3">
<h3 id="ensemble-models">Ensemble Models<a class="anchor" aria-label="anchor" href="#ensemble-models"></a>
</h3>
<p>Combine multiple prediction methods for improved accuracy:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create mock data for demonstration</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></span>
<span><span class="va">pathway_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">30</span> <span class="op">*</span> <span class="fl">10</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">30</span>, ncol <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">pathway_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Pathway"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pathway_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span></span>
<span></span>
<span><span class="va">signature_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fl">10</span> <span class="op">*</span> <span class="fl">3</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">10</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">signature_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Pathway"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">signature_matrix</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"C"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run multiple prediction methods</span></span>
<span><span class="va">similarity_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predict_by_similarity.html">predict_by_similarity</a></span><span class="op">(</span></span>
<span>    pathway_matrix <span class="op">=</span> <span class="va">pathway_matrix</span>,</span>
<span>    signature_matrix <span class="op">=</span> <span class="va">signature_matrix</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">svm_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predict_by_svm.html">predict_by_svm</a></span><span class="op">(</span></span>
<span>    pathway_matrix <span class="op">=</span> <span class="va">pathway_matrix</span>,</span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create an ensemble prediction</span></span>
<span><span class="va">ensemble_predictions</span> <span class="op">&lt;-</span> <span class="fu">create_ensemble_predictions</span><span class="op">(</span></span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>    similarity_results <span class="op">=</span> <span class="va">similarity_results</span>,</span>
<span>    svm_results <span class="op">=</span> <span class="va">svm_results</span>,</span>
<span>    weights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>similarity <span class="op">=</span> <span class="fl">0.4</span>, svm <span class="op">=</span> <span class="fl">0.6</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="custom-svm-parameters">Custom SVM Parameters<a class="anchor" aria-label="anchor" href="#custom-svm-parameters"></a>
</h3>
<p>Fine-tune SVM parameters for optimal performance:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create custom SVM prediction with optimized parameters</span></span>
<span><span class="va">custom_svm_results</span> <span class="op">&lt;-</span> <span class="fu">predict_by_svm_custom</span><span class="op">(</span></span>
<span>    pathway_matrix <span class="op">=</span> <span class="va">pathway_matrix</span>,</span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>    kernel <span class="op">=</span> <span class="st">"radial"</span>,</span>
<span>    cost <span class="op">=</span> <span class="fl">10</span>,</span>
<span>    gamma <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>    probability <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    cross_validation <span class="op">=</span> <span class="fl">5</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="advanced-evaluation">Advanced Evaluation<a class="anchor" aria-label="anchor" href="#advanced-evaluation"></a>
</h2>
<div class="section level3">
<h3 id="cross-validation-evaluation">Cross-validation Evaluation<a class="anchor" aria-label="anchor" href="#cross-validation-evaluation"></a>
</h3>
<p>Perform k-fold cross-validation for robust evaluation:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Perform k-fold cross-validation</span></span>
<span><span class="va">k</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">cv_results</span> <span class="op">&lt;-</span> <span class="fu">cross_validate_predictions</span><span class="op">(</span></span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>    pathway_matrix <span class="op">=</span> <span class="va">pathway_matrix</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"svm"</span>,</span>
<span>    k_folds <span class="op">=</span> <span class="va">k</span>,</span>
<span>    stratify_by <span class="op">=</span> <span class="st">"sample"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize cross-validation results</span></span>
<span><span class="va">cv_plot</span> <span class="op">&lt;-</span> <span class="fu">plot_cross_validation_results</span><span class="op">(</span></span>
<span>    cv_results <span class="op">=</span> <span class="va">cv_results</span>,</span>
<span>    metric <span class="op">=</span> <span class="st">"accuracy"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"5-fold Cross-validation Results"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">cv_plot</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="advanced-metrics">Advanced Metrics<a class="anchor" aria-label="anchor" href="#advanced-metrics"></a>
</h3>
<p>Calculate additional performance metrics:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Calculate advanced metrics</span></span>
<span><span class="va">advanced_metrics</span> <span class="op">&lt;-</span> <span class="fu">calculate_advanced_metrics</span><span class="op">(</span></span>
<span>    true_labels <span class="op">=</span> <span class="va">seurat_object</span><span class="op">$</span><span class="va">sample</span>,</span>
<span>    predicted_labels <span class="op">=</span> <span class="va">svm_results</span><span class="op">$</span><span class="va">predictions</span>,</span>
<span>    predicted_probs <span class="op">=</span> <span class="va">svm_results</span><span class="op">$</span><span class="va">probabilities</span>,</span>
<span>    metrics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"auc"</span>, <span class="st">"precision"</span>, <span class="st">"recall"</span>, <span class="st">"f1"</span>, <span class="st">"kappa"</span>, <span class="st">"mcc"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create detailed metrics visualization</span></span>
<span><span class="co"># The create_evaluation_metrics_plot function is now in inst/extras/alternative_implementations.R</span></span>
<span><span class="co"># metrics_plot &lt;- create_evaluation_metrics_plot(</span></span>
<span><span class="co">#     evaluation_results = advanced_metrics,</span></span>
<span><span class="co">#     plot_type = "metrics",</span></span>
<span><span class="co">#     title = "Advanced Evaluation Metrics"</span></span>
<span><span class="co"># )</span></span>
<span><span class="co"># print(metrics_plot)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="integration-with-other-tools">Integration with Other Tools<a class="anchor" aria-label="anchor" href="#integration-with-other-tools"></a>
</h2>
<div class="section level3">
<h3 id="integration-with-singler">Integration with SingleR<a class="anchor" aria-label="anchor" href="#integration-with-singler"></a>
</h3>
<p>Integrate with SingleR for reference-based cell type annotation:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load required packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleR</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">celldex</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get reference dataset</span></span>
<span><span class="va">ref</span> <span class="op">&lt;-</span> <span class="fu">celldex</span><span class="fu">::</span><span class="fu">HumanPrimaryCellAtlasData</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run SingleR</span></span>
<span><span class="va">singler_results</span> <span class="op">&lt;-</span> <span class="fu">SingleR</span><span class="op">(</span></span>
<span>    test <span class="op">=</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">seurat_object</span>, slot <span class="op">=</span> <span class="st">"data"</span><span class="op">)</span>,</span>
<span>    ref <span class="op">=</span> <span class="va">ref</span>,</span>
<span>    labels <span class="op">=</span> <span class="va">ref</span><span class="op">$</span><span class="va">label.main</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add SingleR annotations to Seurat object</span></span>
<span><span class="va">seurat_object</span><span class="op">$</span><span class="va">singler_labels</span> <span class="op">&lt;-</span> <span class="va">singler_results</span><span class="op">$</span><span class="va">labels</span></span>
<span></span>
<span><span class="co"># Compare SingleR annotations with predicted conditions</span></span>
<span><span class="va">comparison_plot</span> <span class="op">&lt;-</span> <span class="fu">create_annotation_comparison_plot</span><span class="op">(</span></span>
<span>    seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>    singler_column <span class="op">=</span> <span class="st">"singler_labels"</span>,</span>
<span>    prediction_column <span class="op">=</span> <span class="st">"svm_prediction"</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"SingleR vs SVM Predictions"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">comparison_plot</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="integration-with-trajectory-analysis">Integration with Trajectory Analysis<a class="anchor" aria-label="anchor" href="#integration-with-trajectory-analysis"></a>
</h3>
<p>Integrate with trajectory analysis tools:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load required packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">slingshot</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">tradeSeq</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run trajectory analysis</span></span>
<span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span></span>
<span><span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">seurat_object</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get UMAP embedding</span></span>
<span><span class="va">umap_embedding</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.github.io/seurat-object/reference/Embeddings.html" class="external-link">Embeddings</a></span><span class="op">(</span><span class="va">seurat_object</span>, <span class="st">"umap"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get cluster info</span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="va">seurat_object</span><span class="op">$</span><span class="va">seurat_clusters</span></span>
<span></span>
<span><span class="co"># Run Slingshot</span></span>
<span><span class="va">slingshot_results</span> <span class="op">&lt;-</span> <span class="fu">slingshot</span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">umap_embedding</span>,</span>
<span>    clusterLabels <span class="op">=</span> <span class="va">clusters</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize trajectory</span></span>
<span><span class="va">plot_trajectory</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">umap_embedding</span>, <span class="va">slingshot_results</span>, <span class="va">color_by</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">plot_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>        UMAP_1 <span class="op">=</span> <span class="va">umap_embedding</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>,</span>
<span>        UMAP_2 <span class="op">=</span> <span class="va">umap_embedding</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>,</span>
<span>        color <span class="op">=</span> <span class="va">color_by</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span>    <span class="co"># Create plot</span></span>
<span>    <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu">ggplot</span><span class="op">(</span><span class="va">plot_df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">UMAP_1</span>, y <span class="op">=</span> <span class="va">UMAP_2</span>, color <span class="op">=</span> <span class="va">color</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu">geom_point</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu">labs</span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Trajectory Analysis"</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># Add trajectory lines</span></span>
<span>    <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">slingshot_results</span><span class="op">@</span><span class="va">curves</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="va">curve_points</span> <span class="op">&lt;-</span> <span class="va">slingshot_results</span><span class="op">@</span><span class="va">curves</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">s</span><span class="op">[</span><span class="va">slingshot_results</span><span class="op">@</span><span class="va">curves</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">ord</span>, <span class="op">]</span></span>
<span>        <span class="va">curve_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>            UMAP_1 <span class="op">=</span> <span class="va">curve_points</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>,</span>
<span>            UMAP_2 <span class="op">=</span> <span class="va">curve_points</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span></span>
<span>        <span class="op">)</span></span>
<span>        <span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> <span class="fu">geom_path</span><span class="op">(</span></span>
<span>            data <span class="op">=</span> <span class="va">curve_df</span>, <span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">UMAP_1</span>, y <span class="op">=</span> <span class="va">UMAP_2</span><span class="op">)</span>,</span>
<span>            color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">1</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Create trajectory plot</span></span>
<span><span class="va">traj_plot</span> <span class="op">&lt;-</span> <span class="fu">plot_trajectory</span><span class="op">(</span></span>
<span>    umap_embedding <span class="op">=</span> <span class="va">umap_embedding</span>,</span>
<span>    slingshot_results <span class="op">=</span> <span class="va">slingshot_results</span>,</span>
<span>    color_by <span class="op">=</span> <span class="va">seurat_object</span><span class="op">$</span><span class="va">sample</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">traj_plot</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="pipeline-customization">Pipeline Customization<a class="anchor" aria-label="anchor" href="#pipeline-customization"></a>
</h2>
<div class="section level3">
<h3 id="creating-a-custom-analysis-pipeline">Creating a Custom Analysis Pipeline<a class="anchor" aria-label="anchor" href="#creating-a-custom-analysis-pipeline"></a>
</h3>
<p>Build a customized analysis pipeline that combines multiple
steps:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define custom pipeline function</span></span>
<span><span class="va">custom_analysis_pipeline</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span></span>
<span>    <span class="va">tenx_data_dir</span>,</span>
<span>    <span class="va">experiment_id</span>,</span>
<span>    <span class="va">kegg_file</span>,</span>
<span>    <span class="va">output_dir</span> <span class="op">=</span> <span class="st">"results"</span>,</span>
<span>    <span class="va">use_parallel</span> <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    <span class="va">num_cores</span> <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    <span class="va">verbose</span> <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># 1. Load data</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">verbose</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"1. Loading data..."</span><span class="op">)</span></span>
<span>    <span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_10x_data.html">load_10x_data</a></span><span class="op">(</span></span>
<span>        tenx_data_dir <span class="op">=</span> <span class="va">tenx_data_dir</span>,</span>
<span>        experiment_id <span class="op">=</span> <span class="va">experiment_id</span>,</span>
<span>        verbose <span class="op">=</span> <span class="va">verbose</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span>    <span class="co"># 2. Preprocess data</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">verbose</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"2. Preprocessing data..."</span><span class="op">)</span></span>
<span>    <span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocess_data.html">preprocess_data</a></span><span class="op">(</span></span>
<span>        seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>        verbose <span class="op">=</span> <span class="va">verbose</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span>    <span class="co"># 3. Reduce dimensions</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">verbose</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"3. Reducing dimensions..."</span><span class="op">)</span></span>
<span>    <span class="va">seurat_object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/reduce_dimensions.html">reduce_dimensions</a></span><span class="op">(</span></span>
<span>        seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>        verbose <span class="op">=</span> <span class="va">verbose</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span>    <span class="co"># 4. Perform pathway analysis</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">verbose</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"4. Performing pathway analysis..."</span><span class="op">)</span></span>
<span>    <span class="va">kegg_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/parse_kegg_keg.html">parse_kegg_keg</a></span><span class="op">(</span></span>
<span>        kegg_file <span class="op">=</span> <span class="va">kegg_file</span>,</span>
<span>        verbose <span class="op">=</span> <span class="va">verbose</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span>    <span class="va">pathway_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_fingerprints.html">build_fingerprints</a></span><span class="op">(</span></span>
<span>        seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>        kegg_data <span class="op">=</span> <span class="va">kegg_data</span>,</span>
<span>        use_parallel <span class="op">=</span> <span class="va">use_parallel</span>,</span>
<span>        num_cores <span class="op">=</span> <span class="va">num_cores</span>,</span>
<span>        verbose <span class="op">=</span> <span class="va">verbose</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span>    <span class="co"># 5. Create signature matrix</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">verbose</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"5. Creating signature matrix..."</span><span class="op">)</span></span>
<span>    <span class="va">signature_matrix</span> <span class="op">&lt;-</span> <span class="fu">create_signature_matrix</span><span class="op">(</span></span>
<span>        seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>        pathway_matrix <span class="op">=</span> <span class="va">pathway_matrix</span>,</span>
<span>        verbose <span class="op">=</span> <span class="va">verbose</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span>    <span class="co"># 6. Make predictions</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">verbose</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"6. Making predictions..."</span><span class="op">)</span></span>
<span>    <span class="va">similarity_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predict_by_similarity.html">predict_by_similarity</a></span><span class="op">(</span></span>
<span>        pathway_matrix <span class="op">=</span> <span class="va">pathway_matrix</span>,</span>
<span>        signature_matrix <span class="op">=</span> <span class="va">signature_matrix</span>,</span>
<span>        verbose <span class="op">=</span> <span class="va">verbose</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span>    <span class="va">svm_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predict_by_svm.html">predict_by_svm</a></span><span class="op">(</span></span>
<span>        pathway_matrix <span class="op">=</span> <span class="va">pathway_matrix</span>,</span>
<span>        seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>        verbose <span class="op">=</span> <span class="va">verbose</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span>    <span class="co"># 7. Evaluate predictions</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">verbose</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"7. Evaluating predictions..."</span><span class="op">)</span></span>
<span>    <span class="va">evaluation_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/evaluate_predictions.html">evaluate_predictions</a></span><span class="op">(</span><span class="va">seurat_object</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># 8. Create visualizations</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">verbose</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"8. Creating visualizations..."</span><span class="op">)</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.exists</a></span><span class="op">(</span><span class="va">output_dir</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">output_dir</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># Save plots using ggplot2::ggsave</span></span>
<span>    <span class="co"># Note: save_visualization_plots has been moved to inst/extras/alternative_implementations.R</span></span>
<span></span>
<span>    <span class="co"># Save evaluation plots if they exist</span></span>
<span>    <span class="va">evaluation_plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_evaluation_plots.html">create_evaluation_plots</a></span><span class="op">(</span></span>
<span>        seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>        results_dir <span class="op">=</span> <span class="va">output_dir</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span>    <span class="co"># Save individual plots</span></span>
<span>    <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">evaluation_plots</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="va">filename</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">output_dir</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">experiment_name</span>, <span class="st">"_plot_"</span>, <span class="va">i</span>, <span class="st">".png"</span><span class="op">)</span><span class="op">)</span></span>
<span>        <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="va">filename</span>, <span class="va">evaluation_plots</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>, width <span class="op">=</span> <span class="fl">10</span>, height <span class="op">=</span> <span class="fl">8</span>, dpi <span class="op">=</span> <span class="fl">300</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span></span>
<span>    <span class="co"># 9. Return results</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        seurat_object <span class="op">=</span> <span class="va">seurat_object</span>,</span>
<span>        pathway_matrix <span class="op">=</span> <span class="va">pathway_matrix</span>,</span>
<span>        signature_matrix <span class="op">=</span> <span class="va">signature_matrix</span>,</span>
<span>        similarity_results <span class="op">=</span> <span class="va">similarity_results</span>,</span>
<span>        svm_results <span class="op">=</span> <span class="va">svm_results</span>,</span>
<span>        evaluation_results <span class="op">=</span> <span class="va">evaluation_results</span></span>
<span>    <span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Example usage (not run)</span></span>
<span><span class="co"># results &lt;- custom_analysis_pipeline(</span></span>
<span><span class="co">#   tenx_data_dir = tenx_data_dir,</span></span>
<span><span class="co">#   experiment_id = "example",</span></span>
<span><span class="co">#   kegg_file = system.file("extdata", "kegg", "example_pathway.keg", package = "scCulturePredict"),</span></span>
<span><span class="co">#   output_dir = "custom_results",</span></span>
<span><span class="co">#   use_parallel = TRUE,</span></span>
<span><span class="co">#   num_cores = 4,</span></span>
<span><span class="co">#   verbose = TRUE</span></span>
<span><span class="co"># )</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="advanced-singlecellexperiment-support">Advanced SingleCellExperiment Support<a class="anchor" aria-label="anchor" href="#advanced-singlecellexperiment-support"></a>
</h2>
<p>The <code>scCulturePredict</code> package provides comprehensive
support for SingleCellExperiment (SCE) objects, enabling seamless
integration with Bioconductor workflows.</p>
<div class="section level3">
<h3 id="working-with-sce-objects">Working with SCE Objects<a class="anchor" aria-label="anchor" href="#working-with-sce-objects"></a>
</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create or load an SCE object</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span></span>
<span>    assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Poisson.html" class="external-link">rpois</a></span><span class="op">(</span><span class="fl">10000</span>, lambda <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">1000</span>, ncol <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    colData <span class="op">=</span> <span class="fu">DataFrame</span><span class="op">(</span></span>
<span>        sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="st">"Treatment"</span><span class="op">)</span>, <span class="fl">5</span><span class="op">)</span>,</span>
<span>        batch <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Batch1"</span>, <span class="st">"Batch2"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run analysis with SCE object directly</span></span>
<span><span class="va">results_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scCulture.html">scCulture</a></span><span class="op">(</span></span>
<span>    sce_data_path <span class="op">=</span> <span class="va">sce</span>,  <span class="co"># Pass object directly</span></span>
<span>    input_type <span class="op">=</span> <span class="st">"sce"</span>,</span>
<span>    kegg_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kegg"</span>, <span class="st">"example_pathway.keg"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span>,</span>
<span>    output_dir <span class="op">=</span> <span class="st">"./sce_results"</span>,</span>
<span>    mode <span class="op">=</span> <span class="st">"build"</span>,</span>
<span>    experiment_id <span class="op">=</span> <span class="st">"sce_advanced"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Or load from RDS file</span></span>
<span><span class="va">results_sce_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scCulture.html">scCulture</a></span><span class="op">(</span></span>
<span>    sce_data_path <span class="op">=</span> <span class="st">"path/to/sce_data.rds"</span>,</span>
<span>    input_type <span class="op">=</span> <span class="st">"sce"</span>,</span>
<span>    kegg_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kegg"</span>, <span class="st">"example_pathway.keg"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span>,</span>
<span>    output_dir <span class="op">=</span> <span class="st">"./sce_file_results"</span>,</span>
<span>    mode <span class="op">=</span> <span class="st">"build"</span>,</span>
<span>    experiment_id <span class="op">=</span> <span class="st">"sce_from_file"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cross-dataset-predictions">Cross-Dataset Predictions<a class="anchor" aria-label="anchor" href="#cross-dataset-predictions"></a>
</h3>
<p>One of the most powerful features is the ability to train models on
one data format and apply them to another:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Scenario 1: Train on 10X, predict on SCE</span></span>
<span><span class="co"># This is useful when you have labeled 10X reference data</span></span>
<span><span class="co"># and unlabeled SCE query data</span></span>
<span></span>
<span><span class="co"># Step 1: Build model with 10X data</span></span>
<span><span class="va">model_10x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scCulture.html">scCulture</a></span><span class="op">(</span></span>
<span>    tenx_data_dir <span class="op">=</span> <span class="st">"reference_10x_data/"</span>,</span>
<span>    input_type <span class="op">=</span> <span class="st">"10x"</span>,</span>
<span>    kegg_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kegg"</span>, <span class="st">"example_pathway.keg"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span>,</span>
<span>    output_dir <span class="op">=</span> <span class="st">"./model_10x"</span>,</span>
<span>    mode <span class="op">=</span> <span class="st">"build"</span>,</span>
<span>    experiment_id <span class="op">=</span> <span class="st">"10x_reference"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 2: Apply to SCE data</span></span>
<span><span class="va">predictions_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scCulture.html">scCulture</a></span><span class="op">(</span></span>
<span>    sce_data_path <span class="op">=</span> <span class="va">sce</span>,  <span class="co"># or path to RDS file</span></span>
<span>    input_type <span class="op">=</span> <span class="st">"sce"</span>,</span>
<span>    output_dir <span class="op">=</span> <span class="st">"./predictions_on_sce"</span>,</span>
<span>    mode <span class="op">=</span> <span class="st">"predict"</span>,</span>
<span>    fingerprint_file <span class="op">=</span> <span class="va">model_10x</span><span class="op">$</span><span class="va">fingerprint_file</span>,</span>
<span>    experiment_id <span class="op">=</span> <span class="st">"sce_query"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Scenario 2: Train on SCE, predict on 10X</span></span>
<span><span class="co"># Useful when your reference data is in SCE format</span></span>
<span></span>
<span><span class="co"># Step 1: Build model with SCE data</span></span>
<span><span class="va">model_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scCulture.html">scCulture</a></span><span class="op">(</span></span>
<span>    sce_data_path <span class="op">=</span> <span class="st">"reference_sce.rds"</span>,</span>
<span>    input_type <span class="op">=</span> <span class="st">"sce"</span>,</span>
<span>    kegg_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kegg"</span>, <span class="st">"example_pathway.keg"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span>,</span>
<span>    output_dir <span class="op">=</span> <span class="st">"./model_sce"</span>,</span>
<span>    mode <span class="op">=</span> <span class="st">"build"</span>,</span>
<span>    experiment_id <span class="op">=</span> <span class="st">"sce_reference"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Step 2: Apply to 10X data</span></span>
<span><span class="va">predictions_10x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scCulture.html">scCulture</a></span><span class="op">(</span></span>
<span>    tenx_data_dir <span class="op">=</span> <span class="st">"query_10x_data/"</span>,</span>
<span>    input_type <span class="op">=</span> <span class="st">"10x"</span>,</span>
<span>    output_dir <span class="op">=</span> <span class="st">"./predictions_on_10x"</span>,</span>
<span>    mode <span class="op">=</span> <span class="st">"predict"</span>,</span>
<span>    fingerprint_file <span class="op">=</span> <span class="va">model_sce</span><span class="op">$</span><span class="va">fingerprint_file</span>,</span>
<span>    experiment_id <span class="op">=</span> <span class="st">"10x_query"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="batch-processing-with-mixed-data-types">Batch Processing with Mixed Data Types<a class="anchor" aria-label="anchor" href="#batch-processing-with-mixed-data-types"></a>
</h3>
<p>Process multiple datasets of different formats in a single
pipeline:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define datasets with their types</span></span>
<span><span class="va">datasets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"dataset1"</span>, path <span class="op">=</span> <span class="st">"data/10x_1/"</span>, type <span class="op">=</span> <span class="st">"10x"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"dataset2"</span>, path <span class="op">=</span> <span class="st">"data/sce_1.rds"</span>, type <span class="op">=</span> <span class="st">"sce"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"dataset3"</span>, path <span class="op">=</span> <span class="st">"data/10x_2/"</span>, type <span class="op">=</span> <span class="st">"10x"</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>name <span class="op">=</span> <span class="st">"dataset4"</span>, path <span class="op">=</span> <span class="st">"data/sce_2.rds"</span>, type <span class="op">=</span> <span class="st">"sce"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Process all datasets</span></span>
<span><span class="va">results_mixed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">dataset</span> <span class="kw">in</span> <span class="va">datasets</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">dataset</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"10x"</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="va">results_mixed</span><span class="op">[[</span><span class="va">dataset</span><span class="op">$</span><span class="va">name</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scCulture.html">scCulture</a></span><span class="op">(</span></span>
<span>            tenx_data_dir <span class="op">=</span> <span class="va">dataset</span><span class="op">$</span><span class="va">path</span>,</span>
<span>            input_type <span class="op">=</span> <span class="st">"10x"</span>,</span>
<span>            kegg_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kegg"</span>, <span class="st">"example_pathway.keg"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span>,</span>
<span>            output_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"./results_"</span>, <span class="va">dataset</span><span class="op">$</span><span class="va">name</span><span class="op">)</span>,</span>
<span>            mode <span class="op">=</span> <span class="st">"build"</span>,</span>
<span>            experiment_id <span class="op">=</span> <span class="va">dataset</span><span class="op">$</span><span class="va">name</span>,</span>
<span>            verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">}</span> <span class="kw">else</span> <span class="kw">if</span> <span class="op">(</span><span class="va">dataset</span><span class="op">$</span><span class="va">type</span> <span class="op">==</span> <span class="st">"sce"</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="va">results_mixed</span><span class="op">[[</span><span class="va">dataset</span><span class="op">$</span><span class="va">name</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scCulture.html">scCulture</a></span><span class="op">(</span></span>
<span>            sce_data_path <span class="op">=</span> <span class="va">dataset</span><span class="op">$</span><span class="va">path</span>,</span>
<span>            input_type <span class="op">=</span> <span class="st">"sce"</span>,</span>
<span>            kegg_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kegg"</span>, <span class="st">"example_pathway.keg"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span>,</span>
<span>            output_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"./results_"</span>, <span class="va">dataset</span><span class="op">$</span><span class="va">name</span><span class="op">)</span>,</span>
<span>            mode <span class="op">=</span> <span class="st">"build"</span>,</span>
<span>            experiment_id <span class="op">=</span> <span class="va">dataset</span><span class="op">$</span><span class="va">name</span>,</span>
<span>            verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Compare results across datasets</span></span>
<span><span class="va">accuracy_comparison</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">results_mixed</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">x</span><span class="op">$</span><span class="va">evaluation_results</span><span class="op">$</span><span class="va">overall_accuracy</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">accuracy_comparison</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="advanced-sce-integration">Advanced SCE Integration<a class="anchor" aria-label="anchor" href="#advanced-sce-integration"></a>
</h3>
<p>Leverage SCE-specific features while using scCulturePredict:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Use existing dimensionality reductions from SCE</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scater</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Prepare SCE with existing analyses</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">logNormCounts</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">runPCA</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">runUMAP</span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The package will automatically detect and use these reductions</span></span>
<span><span class="va">results_with_reductions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scCulture.html">scCulture</a></span><span class="op">(</span></span>
<span>    sce_data_path <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    input_type <span class="op">=</span> <span class="st">"sce"</span>,</span>
<span>    kegg_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"kegg"</span>, <span class="st">"example_pathway.keg"</span>, package <span class="op">=</span> <span class="st">"scCulturePredict"</span><span class="op">)</span>,</span>
<span>    output_dir <span class="op">=</span> <span class="st">"./sce_with_reductions"</span>,</span>
<span>    mode <span class="op">=</span> <span class="st">"build"</span>,</span>
<span>    experiment_id <span class="op">=</span> <span class="st">"sce_integrated"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Access the results as Seurat object for further analysis</span></span>
<span><span class="va">seurat_from_sce</span> <span class="op">&lt;-</span> <span class="va">results_with_reductions</span><span class="op">$</span><span class="va">seurat_object</span></span>
<span></span>
<span><span class="co"># Or convert back to SCE if needed for downstream Bioconductor workflows</span></span>
<span><span class="co"># (requires additional conversion code)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>This vignette demonstrated advanced usage scenarios for the
<code>scCulturePredict</code> package, including batch effect
correction, custom pathway analysis, performance optimization, advanced
prediction models, integration with other single-cell analysis tools,
and comprehensive SingleCellExperiment support. These techniques can
help you extract more insights from your single-cell data and customize
the analysis to your specific research questions.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] dplyr_1.1.4              Seurat_5.3.0             SeuratObject_5.1.0      </span></span>
<span><span class="co">#&gt; [4] sp_2.2-0                 scCulturePredict_0.99.30 BiocStyle_2.36.0        </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3          jsonlite_2.0.0             </span></span>
<span><span class="co">#&gt;   [3] magrittr_2.0.3              spatstat.utils_3.1-5       </span></span>
<span><span class="co">#&gt;   [5] farver_2.1.2                rmarkdown_2.29             </span></span>
<span><span class="co">#&gt;   [7] fs_1.6.6                    ragg_1.4.0                 </span></span>
<span><span class="co">#&gt;   [9] vctrs_0.6.5                 ROCR_1.0-11                </span></span>
<span><span class="co">#&gt;  [11] spatstat.explore_3.5-2      htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt;  [13] S4Arrays_1.8.1              SparseArray_1.8.1          </span></span>
<span><span class="co">#&gt;  [15] sass_0.4.10                 sctransform_0.4.2          </span></span>
<span><span class="co">#&gt;  [17] parallelly_1.45.1           KernSmooth_2.23-26         </span></span>
<span><span class="co">#&gt;  [19] bslib_0.9.0                 htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt;  [21] desc_1.4.3                  ica_1.0-3                  </span></span>
<span><span class="co">#&gt;  [23] plyr_1.8.9                  plotly_4.11.0              </span></span>
<span><span class="co">#&gt;  [25] zoo_1.8-14                  cachem_1.1.0               </span></span>
<span><span class="co">#&gt;  [27] igraph_2.1.4                mime_0.13                  </span></span>
<span><span class="co">#&gt;  [29] lifecycle_1.0.4             pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [31] Matrix_1.7-3                R6_2.6.1                   </span></span>
<span><span class="co">#&gt;  [33] fastmap_1.2.0               GenomeInfoDbData_1.2.14    </span></span>
<span><span class="co">#&gt;  [35] MatrixGenerics_1.20.0       fitdistrplus_1.2-4         </span></span>
<span><span class="co">#&gt;  [37] future_1.67.0               shiny_1.11.1               </span></span>
<span><span class="co">#&gt;  [39] digest_0.6.37               patchwork_1.3.2            </span></span>
<span><span class="co">#&gt;  [41] S4Vectors_0.46.0            tensor_1.5.1               </span></span>
<span><span class="co">#&gt;  [43] RSpectra_0.16-2             irlba_2.3.5.1              </span></span>
<span><span class="co">#&gt;  [45] textshaping_1.0.1           GenomicRanges_1.60.0       </span></span>
<span><span class="co">#&gt;  [47] progressr_0.15.1            spatstat.sparse_3.1-0      </span></span>
<span><span class="co">#&gt;  [49] polyclip_1.10-7             httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [51] abind_1.4-8                 compiler_4.5.1             </span></span>
<span><span class="co">#&gt;  [53] fastDummies_1.7.5           MASS_7.3-65                </span></span>
<span><span class="co">#&gt;  [55] DelayedArray_0.34.1         tools_4.5.1                </span></span>
<span><span class="co">#&gt;  [57] lmtest_0.9-40               httpuv_1.6.16              </span></span>
<span><span class="co">#&gt;  [59] future.apply_1.20.0         goftest_1.2-3              </span></span>
<span><span class="co">#&gt;  [61] glue_1.8.0                  nlme_3.1-168               </span></span>
<span><span class="co">#&gt;  [63] promises_1.3.3              grid_4.5.1                 </span></span>
<span><span class="co">#&gt;  [65] Rtsne_0.17                  cluster_2.1.8.1            </span></span>
<span><span class="co">#&gt;  [67] reshape2_1.4.4              generics_0.1.4             </span></span>
<span><span class="co">#&gt;  [69] gtable_0.3.6                spatstat.data_3.1-8        </span></span>
<span><span class="co">#&gt;  [71] tidyr_1.3.1                 data.table_1.17.8          </span></span>
<span><span class="co">#&gt;  [73] XVector_0.48.0              spatstat.geom_3.5-0        </span></span>
<span><span class="co">#&gt;  [75] BiocGenerics_0.54.0         RcppAnnoy_0.0.22           </span></span>
<span><span class="co">#&gt;  [77] ggrepel_0.9.6               RANN_2.6.2                 </span></span>
<span><span class="co">#&gt;  [79] pillar_1.11.0               stringr_1.5.1              </span></span>
<span><span class="co">#&gt;  [81] spam_2.11-1                 RcppHNSW_0.6.0             </span></span>
<span><span class="co">#&gt;  [83] later_1.4.3                 splines_4.5.1              </span></span>
<span><span class="co">#&gt;  [85] lattice_0.22-7              deldir_2.0-4               </span></span>
<span><span class="co">#&gt;  [87] survival_3.8-3              tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [89] SingleCellExperiment_1.30.1 miniUI_0.1.2               </span></span>
<span><span class="co">#&gt;  [91] pbapply_1.7-4               knitr_1.50                 </span></span>
<span><span class="co">#&gt;  [93] gridExtra_2.3               bookdown_0.44              </span></span>
<span><span class="co">#&gt;  [95] IRanges_2.42.0              SummarizedExperiment_1.38.1</span></span>
<span><span class="co">#&gt;  [97] scattermore_1.2             stats4_4.5.1               </span></span>
<span><span class="co">#&gt;  [99] xfun_0.53                   Biobase_2.68.0             </span></span>
<span><span class="co">#&gt; [101] matrixStats_1.5.0           stringi_1.8.7              </span></span>
<span><span class="co">#&gt; [103] UCSC.utils_1.4.0            lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt; [105] yaml_2.3.10                 evaluate_1.0.4             </span></span>
<span><span class="co">#&gt; [107] codetools_0.2-20            tibble_3.3.0               </span></span>
<span><span class="co">#&gt; [109] BiocManager_1.30.26         cli_3.6.5                  </span></span>
<span><span class="co">#&gt; [111] uwot_0.2.3                  xtable_1.8-4               </span></span>
<span><span class="co">#&gt; [113] reticulate_1.43.0           systemfonts_1.2.3          </span></span>
<span><span class="co">#&gt; [115] jquerylib_0.1.4             Rcpp_1.1.0                 </span></span>
<span><span class="co">#&gt; [117] GenomeInfoDb_1.44.2         spatstat.random_3.4-1      </span></span>
<span><span class="co">#&gt; [119] globals_0.18.0              png_0.1-8                  </span></span>
<span><span class="co">#&gt; [121] spatstat.univar_3.1-4       parallel_4.5.1             </span></span>
<span><span class="co">#&gt; [123] pkgdown_2.1.3               ggplot2_3.5.2              </span></span>
<span><span class="co">#&gt; [125] dotCall64_1.2               listenv_0.9.1              </span></span>
<span><span class="co">#&gt; [127] viridisLite_0.4.2           scales_1.4.0               </span></span>
<span><span class="co">#&gt; [129] ggridges_0.5.6              purrr_1.1.0                </span></span>
<span><span class="co">#&gt; [131] crayon_1.5.3                rlang_1.1.6                </span></span>
<span><span class="co">#&gt; [133] cowplot_1.2.0</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Niccolò Bianchi.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
